src/ir/localize.h

```diff
@@ -44,13 +44,20 @@ struct Localizer {
   }
 };
 
-// Replaces all children with gets of locals, if they have any effects. After
-// this, the original input has only local.gets as inputs, or other things that
-// have no interacting effects, and so those children can be reordered.
+// Replaces all children with gets of locals, if they have any effects that
+// interact with any of the others, or if they have side effects which cannot be
+// removed.
+//
+// After this, the original input has only local.gets as inputs, or other things
+// that have no interacting effects, and so those children can be reordered
+// and/or removed as needed.
```

Comment:
So is the objective of this class is make all children of an expression freely reorderable? What use cases do you imagine for this other than field removal?

Reply:
For example when we turn an if into a select or vice versa then the condition moves around.

Also, removal and reordering are related, since if we want to remove a field from an instruction but it has effects then we need to move it to another position (like a drop either before or after), which can require reordering with the other children.

In general this pattern of "replace an instruction with a new one using a modified subset of the children" is common in OptimizeInstructions and RemoveUnusedBrs, and hopefully we can generalize and share that code.